README.md

```diff
@@ -20,3 +20,11 @@ randomly without memory instead of systematically trying them?
 
 Describe your reasoning and the conclusion you've come to. Your reasoning is the
 most important part. Add your answer to this markdown file.
+
+The runtime complexity of the algorithm I implemented is $O(n! * n)$. This is because the algorithm generates all permutations of an array of length $n$, which takes $O(n!)$, and then checks each permutation to see if it is sorted. The `isSorted` function has a complexity of $O(n)$, so checking all permutations results in a total complexity of $O(n! * n)$.In the best case, where the input array is already sorted, the algorithm only needs to generate and check the first permutation. This reduces the complexity to $O(n)$ since no additional permutations are needed. In the worst case, such as when the array is in reverse order, the algorithm must generate and check all $n!$ permutations, resulting in the full $O(n! * n)$ complexity.If permutations were generated randomly without memory instead of systematically, the behavior would change. Each random permutation has a $1/n!$ chance of being sorted, so it could find the sorted array in fewer attempts by chance. However, there’s no guarantee that the sorted permutation would be found within a finite number of trials. On average, the complexity remains $O(n! * n)$, but randomness introduces variability in performance.In summary, the systematic approach guarantees finding the solution but is computationally expensive. Random generation adds variability and may occasionally be faster but lacks any guarantee of success within practical limits.
```

Comment:
Are you not always generating all permutations?

Reply:
It does until generate all permuations till the sorted one is found. So worse case it goes through them all but best case it finds and stops early.

Reply:
Then I guess I don't understand your code. Where does it check whether the permutation corresponds to a sorted array before generating the next permutation?

Reply:
You are right, I am thinking about it wrong and what my initial thought process was. It does go through all permutations no matter the case. It would need another implementation to stop at the sorted case.